src/test/java/bio/terra/service/tabulardata/google/BigQueryPdaoUnitTest.java

```diff
@@ -0,0 +1,749 @@
+                            .region(GoogleRegion.NORTHAMERICA_NORTHEAST1))))
+                    .assetSpecification(new AssetSpecification()
+                        .rootTable(new AssetTable()
+                            .datasetTable(tbl1))
+                        .rootColumn(new AssetColumn()
+                            .datasetTable(tbl1)
+                            .datasetColumn(tbl1.getColumns().get(0)))
+                        .assetRelationships(List.of(
+                            new AssetRelationship()
+                                .datasetRelationship(new Relationship()
+                                    .fromTable(tbl1).fromColumn(tbl1.getColumns().get(0))
+                                    .toTable(tbl2).toColumn(tbl2.getColumns().get(0))))))
+                    .snapshotMapTables(List.of(
+                        new SnapshotMapTable().fromTable(tbl1).toTable(snpTbl1)
+                            .snapshotMapColumns(List.of(
+                                new SnapshotMapColumn()
+                                    .fromColumn(tbl1.getColumns().get(0)).toColumn(snpTbl1.getColumns().get(0)),
+                                new SnapshotMapColumn()
+                                    .fromColumn(tbl1.getColumns().get(1)).toColumn(snpTbl1.getColumns().get(1)))),
+                        new SnapshotMapTable().fromTable(tbl2).toTable(snpTbl2)
+                            // Note: purposefully not mapping the third column and swappign the first two\
+                            // for special handling
+                            .snapshotMapColumns(List.of(
+                                new SnapshotMapColumn()
+                                    .fromColumn(tbl2.getColumns().get(0)).toColumn(snpTbl2.getColumns().get(1)),
+                                new SnapshotMapColumn()
+                                    .fromColumn(tbl2.getColumns().get(1)).toColumn(snpTbl2.getColumns().get(0))))))));
+    }
+
+    private void mockNumRowIds(Snapshot snapshot, String tableName, int numRowIds) {
+        String datasetProjectId =
+            snapshot.getFirstSnapshotSource().getDataset().getProjectResource().getGoogleProjectId();
+        String datasetName =
+            snapshot.getFirstSnapshotSource().getDataset().getName();
+        String snapshotProjectId =
+            snapshot.getProjectResource().getGoogleProjectId();
+        String snapshotName = snapshot.getName();
+        BQTestUtils.mockBQQuery(
+            bigQueryProjectSnapshot,
+            "SELECT COUNT(1) " +
+                "FROM `" + datasetProjectId + ".datarepo_" + datasetName + "." + tableName + "` AS T, " +
+                "`" + snapshotProjectId + "." + snapshotName + ".datarepo_row_ids` AS R " +
+                "WHERE R.datarepo_row_id = T.datarepo_row_id",
+            Schema.of(Field.of("val", LegacySQLTypeName.NUMERIC)),
+            List.of(Map.of("val", Integer.toString(numRowIds))));
```

Comment:
In general I prefer `String.valueOf()` since it doesn't matter what type its argument is, and doesn't need to be changed if an underlying type changes.
```suggestion
            List.of(Map.of("val", String.valueOf(numRowIds))));
```